### PR TITLE
update for gcc10 compilation

### PR DESCRIPTION
**Pull request by faizol**
Update data_type.h and type_set.h to be compilable with gcc-10

### CODE REVIEW

**Comment by ZaMaZaN4iK** [comment]
I think that's not proper fix. Since @rbock uses `size_t` everywhere in th eproject - will be better just innclude `stddef.h`

Maybe later @rbock will refactor `size_t` to `std::size_t`

**Comment by ZaMaZaN4iK** [comment]
https://github.com/rbock/sqlpp11/issues/331

File: include/sqlpp11/detail/type_set.h [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #define SQLPP11_DETAIL_TYPE_SET_H
 
 #include <type_traits>
+#include <cstddef>
```
Comment (ZaMaZaN4iK) [inline include/sqlpp11/detail/type_set.h:31]: Use `stddef.h` instead

**Comment by ZaMaZaN4iK** [comment]
Thank you a lot!

**Comment by rbock** [comment]
Thanks  for the cooperation :-)

**Comment by ZaMaZaN4iK** [comment]
@rbock can you make a release of the library? I want to prepare package for your library with fixes. For this I need new release version. Thank you :)

**Comment by rbock** [comment]
Done :-)

**Pull request merged**